R/bayesian.R

```diff
@@ -0,0 +1,311 @@
+#'
+#' Fetch land use results and aggregate to region, commodity, and year.
+#'
+#' @param aScenarioInfo ScenarioInfo structure for the run.
+#' @return Table with region, commodity, year, and area.
+#' @export
+get_scenario_land_data <- function(aScenarioInfo)
+{
+    land.type <- year <- land.allocation <- NULL # silence package notes
+
+    outputdir <- aScenarioInfo$mOutputDir
+    filename <- paste0('landAllocation_',aScenarioInfo$mFileName,'.csv')
+    fn <- file.path(outputdir, 'land', filename)
+
+    readr::read_csv(fn) %>%
+      ## split name / AEZ
+      tidyr::extract('name', c('land.type', 'AEZ'),
+                     '(.+)(AEZ[0-9]+)') %>%
+      dplyr::group_by(land.type, year) %>%
+      dplyr::summarise(model = sum(land.allocation)) %>%
+      ungroup %>%
+      dplyr::mutate(variable = "Land Area") %>%
+      dplyr::mutate(region = aScenarioInfo$mRegion)
+}
+
+
+#' Add model parameter values to a table of model results
+#'
+#' @param df Data frame with model results
+#' @param aScenarioInfo ScenarioInfo structure for the scenario
+#' @keywords internal
+add_parameter_data <- function(df, aScenarioInfo)
+{
+    df$expectation.type <- aScenarioInfo$mExpectationType
+    df$share.old <- aScenarioInfo$mLaggedShareOld
+    df$linear.years <- aScenarioInfo$mLinearYears
+    df$logit.agforest <- aScenarioInfo$mLogitAgroForest
+    df$logit.afnonpast <- aScenarioInfo$mLogitAgroForest_NonPasture
+    df$logit.crop <- aScenarioInfo$mLogitCropland
+    df$region <- aScenarioInfo$mRegion
+
+    df
+}
+
+#' Select a log-probability density function
+#'
+#' Choose a log-probability density function from the family of Gossett's
+#' t-distributions (including the normal distribution as a special case).
+#'
+#' The t-distributions are parameterized by a parameter \eqn{\nu} called the
+#' "degrees of freedom".  Despite the name, this parameter need not be an
+#' integer; however, it must be positive.  The smaller \eqn{\nu} is, the heavier
+#' the tails of the distribution.  In the limit that \eqn{\nu \rightarrow \infty},
+#' the distribution becomes equivalent to the normal distribution.  Therefore,
+#' as a special case, passing \code{df = Inf} will return a normal distribution.
+#'
+#' The function returned from this generator should be called with a vector of
+#' differences between the model data and observed data, and a vector of scale
+#' factors \eqn{sigma}.  Together, these will be used to compute
+#' t-scores; that is, scaled differences between the model data and observed
+#' data: \eqn{t = (M-O)/\sigma}.
+#'
+#' The scaling factor \eqn{\sigma} is a parameter of the probability model.
+#' Since the scales of the observed and model values depend on the commodity
+#' and/or region, this will be a vector of the same length as the difference
+#' vector.
+#'
+#' @param df Degrees of freedom for the distribution.
+#' @return A function \code{lp(model-observation, sigma)}
+#' @export
+get_lpdf <- function(df)
```

Comment:
you use `df` to mean degrees of freedom here, but data frame in the previous function. Maybe spell both out a little bit more.

Reply:
`df` is the name that pretty much all statistical functions use to mean degrees of freedom, so I think it's better to leave that one as is.  I expanded the name of the model data frame in the other function.

Reply:
Perfect.